jupyterlab_bigquery/src/components/details_panel/details_panel.tsx

```diff
@@ -80,18 +83,63 @@ interface SharedDetails {
   labels: string[];
   name: string;
   schema?: SchemaField[];
+  query?: string;
 }
 
 interface Props {
   details: SharedDetails;
   rows: any[];
   // TODO(cxjia): figure out a shared typing for these rows
-  detailsType: string;
+  detailsType: 'DATASET' | 'TABLE' | 'VIEW';
 }
 
+const getTitle = type => {
+  switch (type) {
+    case 'DATASET':
+      return 'Dataset info';
+    case 'TABLE':
+      return 'Table info';
+    case 'VIEW':
+      return 'View info';
+  }
+};
+
 export const DetailsPanel: React.SFC<Props> = props => {
   const { details, rows, detailsType } = props;
 
+  function handleEditorDidMount(_, editor) {
+    const editorElement = editor.getDomNode();
+    if (!editorElement) {
+      return;
+    }
+
+    monaco
+      .init()
+      .then(monaco => {
+        const lineHeight = editor.getOption(
+          monaco.editor.EditorOption.lineHeight
+        );
+        const lineCount = editor._modelData.viewModel.getLineCount();
+        const height = lineCount * lineHeight;
+        editorElement.style.height = `${height}px`;
+        editor.layout();
+
+        monaco.editor.defineTheme('viewQueryTheme', {
+          base: 'vs',
+          inherit: true,
+          rules: [],
+          colors: { 'editorCursor.foreground': '#FFFFFF' },
+        });
+        monaco.editor.setTheme('viewQueryTheme');
```

Comment:
Some hacky stuff here to get a nice view-only editor - there doesn't seem to be a way to disable the cursor so I just made it the same color as the background (white)